File: client/iih_stub/iih.ci_stub/iih.ci.diag/dto/d/DidtoAggDTO.cs
using System;
using xap.mw.core.data;

namespace iih.ci.diag.dto.d
{
	/// <summary>
	/// 诊断headdtoDTO
	/// </summary>
	public class DidtoAggDTO : BaseAggDO
	{	
	    private static string HEADDTO_CIDIDTOZY="iih.ci.diag.dto.d.Cididtozy";
	    private static string HEADDTO_CIDIXY="iih.ci.diag.dto.d.Cidixy";

        public DidtoAggDTO() {
            this.setParentDO(new Headdto());
        }

  	    public Headdto getParentDO() {
		    return ((Headdto) base.ParentDO);
	    }      
	    
	    public void setParentDO(Headdto headDO) {
		    base.ParentDO=headDO;
	    }
	    
	    public Cididtozy[] getCididtozy() {		    
		    
		    BaseDO[] dos=getChildren(Type.GetType(HEADDTO_CIDIDTOZY));
		    if (dos == null || dos.Length==0){return null;}
            Cididtozy[] r = new Cididtozy[dos.Length];
            for (int i = 0; i < dos.Length;i++ ){
                r[i] = dos[i] as Cididtozy;
            }
	        return r;
		    
	    }
	    
	    public void setCididtozy(Cididtozy[] dos) {
            if (dos == null || dos.Length == 0) return;
            setChildren(dos[0].GetType(), dos);
	    }
	    public Cidixy[] getCidixy() {		    
		    
		    BaseDO[] dos=getChildren(Type.GetType(HEADDTO_CIDIXY));
		    if (dos == null || dos.Length==0){return null;}
            Cidixy[] r = new Cidixy[dos.Length];
            for (int i = 0; i < dos.Length;i++ ){
                r[i] = dos[i] as Cidixy;
            }
	        return r;
		    
	    }
	    
	    public void setCidixy(Cidixy[] dos) {
            if (dos == null || dos.Length == 0) return;
            setChildren(dos[0].GetType(), dos);
	    }


	    public override BaseDO createParentDO()
	    {
	        return new Headdto();
	    }

        public override BaseDO createChildDO(string clzName)
	    {
            if (HEADDTO_CIDIDTOZY == clzName)
	        {
                return new Cididtozy();
	        }
           else if (HEADDTO_CIDIXY == clzName)
	        {
                return new Cidixy();
	        }
	        return null;
	    }
	    
	    /// <summary>
        /// 获取子类全路径名数组
        /// </summary>
        /// <returns></returns>
        public override string[] getChildClassNames(){
            return  new string[]{"iih.ci.diag.dto.d.Cididtozy","iih.ci.diag.dto.d.Cidixy"};
        }
	    
	}
}

File: server/iih_pub/iih.ci_pub/src/main/java/iih/ci/diag/dto/d/DidtoAggDTO.java
package iih.ci.diag.dto.d;

import xap.sys.appfw.orm.model.agg.BaseAggDO;
import xap.mw.core.data.BaseDO;
import xap.mw.coreitf.i.IAggDesc;

/**
 * 诊断headdto
 */
public class DidtoAggDTO extends BaseAggDO {
	private static final long serialVersionUID = 1L;

	@Override
	public Headdto getParentDO() {
		return ((Headdto) super.getParentDO());
	}

	public void setParentDO(Headdto headDO) {
		setParent(headDO);
	}

	public Cididtozy[] getCididtozy() {
		return ((Cididtozy[]) getChildren(Cididtozy.class));
	}
	
	public void setCididtozy(Cididtozy[] dos) {
		setChildren(Cididtozy.class, dos);
	}
	public Cidixy[] getCidixy() {
		return ((Cidixy[]) getChildren(Cidixy.class));
	}
	
	public void setCidixy(Cidixy[] dos) {
		setChildren(Cidixy.class, dos);
	}

	 public  BaseDO createParentDO() {
         return new Headdto();
     }

     public  BaseDO createChildDO(String clzName) {
	     if (clzName.equals("iih.ci.diag.dto.d.Cididtozy")) {
             return new Cididtozy();
         }
	    else if (clzName.equals("iih.ci.diag.dto.d.Cidixy")) {
             return new Cidixy();
         }
         return null;
     }
    
    /**
     * AggDTO 中元数据描述不存在，此方法不应使用 
     */ 
    @Override
	public IAggDesc getAggDesc() {
		return null;
	}

}